R/PipeOpPredClassifSurv.R

```diff
@@ -0,0 +1,108 @@
+#' @title PipeOpPredClassifSurv
+#' @name mlr_pipeops_trafopred_classifsurv
+#'
+#' @description
+#' Transform [PredictionClassif] to [PredictionSurv].
+#'
+#' @section Input and Output Channels:
+#' Input and output channels are inherited from [PipeOp].
+#'
+#' The output is the input [PredictionClassif] transformed to a [PredictionSurv].
+#'
+#' @examples
+#' \dontrun{
+#' if (requireNamespace("mlr3pipelines", quietly = TRUE)) {
+#'   library(mlr3)
+#'   library(mlr3pipelines)
+#'
+#'   task = tsk("rats")
+#'
+#'   if (requireNamespace("mlr3learners", quietly = TRUE)) {
+#'     library(mlr3learners)
+#'     po_tasktoclassif = po("trafotask_survclassif")
+#'     po_tasktoclassif$train(list(task))
+#'     task_classif = po_tasktoclassif$predict(list(task))[[1]]
+#'     trafo_data = po_tasktoclassif$predict(list(task))[[2]]
+#'
+#'     learner = lrn("classif.log_reg", predict_type = "prob")
+#'     pred = learner$train(task_classif)$predict_newdata(trafo_data)
+#'
+#'     po_predtosurv = po("trafopred_classifsurv")
+#'     po_predtosurv$train(list(pred, trafo_data))
+#'     po_predtosurv$predict(list(pred, trafo_data))
+#'
+#'   }
+#' }
+#' }
+#' @family PipeOps
+#' @family Transformation PipeOps
+#' @export
+PipeOpPredClassifSurv = R6Class(
+  "PipeOpPredClassifSurv",
+  inherit = mlr3pipelines::PipeOp,
+
+  public = list(
+    #' @description
+    #' Creates a new instance of this [R6][R6::R6Class] class.
+    #' @param id (character(1))\cr
+    #' Identifier of the resulting object.
+    initialize = function(id = "trafopred_classifsurv") {
+      super$initialize(
+        id = id,
+        input = data.table::data.table(
+          name = c("input", "transformed_data"),
+          train = c("NULL", "data.frame"),
+          predict = c("PredictionClassif", "data.frame")
+        ),
+        output = data.table::data.table(
+          name = "output",
+          train = "NULL",
+          predict = "PredictionSurv"
+        )
+      )
+    }
+  ),
+
+  private = list(
+    .predict = function(input) {
+      pred = input[[1]]
+      data = input[[2]]
+      data = cbind(data, pred = pred$prob[, 2])
```

Comment:
maek data a `data.table` from here? or just pass it as data.table from the beggining?